### ISSUE TITLE
False positive 'newline-after-var' with 'semi': never

### ISSUE THREAD

**Issue by wKich** [eslint/eslint#6834]
**What version of ESLint are you using?**
3.2.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
https://github.com/wKich/eslint-config-skbkontur-keforms

**What did you do? Please include the actual source code causing the issue.**
```javascript
const foo = {}

;(foo.bar || []).doSomething()
```

**What did you expect to happen?**
Don't get any warnings or errors.

**What actually happened? Please include the actual, raw output from ESLint.**
Get `newline-after-var` warning

**Comment by vitorbal** [comment]
Thanks for reaching out, @wKich! To help us, could you please post the **raw** output from ESLint when it runs and produces the warning you described?

**Comment by eslintbot** [comment]
Hi @wKich, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by wKich** [comment]
@vitorbal, sorry for didn't post output right away. Here it is
<img width="586" alt="screen shot 2016-08-04 at 10 32 23" src="https://cloud.githubusercontent.com/assets/6397708/17391362/cbc58f3c-5a2e-11e6-81e6-a06f0fe842a9.png">

**Comment by vitorbal** [comment]
Thanks! I can confirm this as well, with the following config and test file:
```json
{
    "root": true,
    "parserOptions": {
        "ecmaVersion": 6
    },
    "rules": {
        "semi": ["error", "never"],
        "newline-after-var": ["error"]
    }
}

```

```javascript
var foo = bar

;(1 || 2).baz()
```

Results in:
```bash
$ npm run lint

> eslint_bug_tester@1.0.0 lint /Users/vitorbalocco/Workspace/eslint_bug_tester
> eslint src/


/Users/vitorbalocco/Workspace/eslint_bug_tester/src/test.js
  1:1  error  Expected blank line after variable declarations  newline-after-var

✖ 1 problem (1 error, 0 warnings)
```

**Comment by mysticatea** [comment]
I will work on this.
